ee/codegen/src/generators/nodes/api-node.ts

```diff
@@ -27,16 +27,50 @@ export class ApiNode extends BaseSingleFileNode<ApiNodeType, ApiNodeContext> {
       })
     );
 
+    const methodInput = this.nodeInputsByKey.get("method");
+    if (!methodInput) {
+      throw new NodeAttributeGenerationError(
+        'Node input "method" is required but not found.'
+      );
+    }
+
+    const methodValue = this.nodeData.inputs
+      .find((input) => input.id === this.nodeData.data.methodInputId)
+      ?.value.rules.find(
+        (value) => value.type === "CONSTANT_VALUE"
+      ) as ConstantValuePointer;
+    if (!methodValue) {
+      throw new NodeAttributeGenerationError(
+        `No method input found for input id ${this.nodeData.data.methodInputId} and of type "CONSTANT_VALUE"`
+      );
+    }
+
+    const isGetMethod = methodValue.data.value === "GET";
+    const body = this.nodeInputsByKey.get("body");
+    const hasBody = !!body;
+    const additionalHeaders = this.nodeData.data.additionalHeaders;
+    const hasHeaders = !!additionalHeaders && additionalHeaders.length > 0;
+    const hasApiKeyHeader = !!this.nodeData.data.apiKeyHeaderKeyInputId;
+
+    // If all four conditions are met (GET, no body, no headers, no API key)
+    // return with just the URL field
+    if (isGetMethod && !hasBody && !hasHeaders && !hasApiKeyHeader) {
+      return [
+        python.field({
+          name: "url",
+          initializer: urlInput,
+        }),
+      ];
+    }
```

Comment:
[blocking] instead of early returning we simply want to iterate over each attribute, and only add to `statements` if it's non default